src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java

```diff
@@ -0,0 +1,667 @@
+package com.minecolonies.coremod.colony.buildings;
+
+import com.minecolonies.api.util.BlockPosUtil;
+import com.minecolonies.api.util.ItemStackUtils;
+import com.minecolonies.blockout.views.Window;
+import com.minecolonies.coremod.achievements.ModAchievements;
+import com.minecolonies.coremod.client.gui.WindowHutGuardTower;
+import com.minecolonies.coremod.colony.CitizenData;
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.colony.ColonyView;
+import com.minecolonies.coremod.colony.jobs.AbstractJob;
+import com.minecolonies.coremod.colony.jobs.JobGuard;
+import com.minecolonies.coremod.entity.EntityCitizen;
+import io.netty.buffer.ByteBuf;
+import net.minecraft.entity.SharedMonsterAttributes;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.*;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraft.nbt.NBTUtil;
+import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.common.util.Constants;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Building class of the guard tower.
+ */
+public abstract class AbstractBuildingGuards extends AbstractBuildingWorker
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Override the "equals" method in this class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2160)